### PR TITLE
HAAR-161: display and filter by gender

### PR DESCRIPTION
**Pull request by neilmendum**
https://dsdmoj.atlassian.net/browse/HAAR-161

### CODE REVIEW

File: server/routes/prisonRegister/prisonRegisterController.ts [hunk 1]
```diff
@@ -29,10 +29,19 @@ export default class PrisonRegisterController {
   }
 
   parseFilter(req: Request): AllPrisonsFilter {
+    let gendersFromQuery = CourtRegisterController.parseStringArrayFromQuery(req.query.genders as string[])
+    if (PrisonRegisterController.doNotFilterIfBothGendersInQuery(gendersFromQuery)) {
+      gendersFromQuery = undefined
```
Comment (neilmendum) [inline server/routes/prisonRegister/prisonRegisterController.ts:34]: We don't need to provide a filter to the backend if the following scenario happens:
1. User clicks female checkbox in order to filter for male only
2. User clicks apply filter
3. User then re-clicks female checkbox

Hence we clear the array before constructing `AllPrisonsFilter`

File: integration_tests/integration/prison-register-list.spec.ts [hunk 2]
```diff
@@ -100,34 +100,88 @@ context('Prison register - prison list navigation', () => {
...
+    // Filter on male prisons only
+    page.femaleFilter().click()
+    page.applyFilterButton().click()
+    page.showFilterButton().click()
+    // Check the male prisons filter has been applied
+    page.maleFilter().should('be.checked')
+    page.femaleFilter().should('not.be.checked')
+    cy.url().should('include', 'genders=MALE')
+  })
+
```
Comment (karenmillermoj) [inline integration_tests/integration/prison-register-list.spec.ts:141]: Should we test for deselecting male and female and test filter values returned.

**Review by karenmillermoj — DISMISSED** [review]
lgtm

File: integration_tests/integration/prison-register-list.spec.ts (hunk 2 above)
Comment (neilmendum) [reply integration_tests/integration/prison-register-list.spec.ts:148]: Good shout. Added that to `Will change the gender filter`

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**